src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -863,6 +863,41 @@ public IEnumerable<SyntaxTree> GetSyntaxTreesWithNameFromDeclarationOnlyCompilat
                 return version;
             }
             #endregion
+
+            private class MetadataReferenceComparer : IEqualityComparer<MetadataReference>
```

Comment:
"CompilationReferencesReferenceSameCompilationComparer"? The behavior is quite specialized, so an equally-specialized name might be appropriate.